src/instructlab/model/train.py

```diff
@@ -87,29 +159,72 @@
         "(reduces GPU VRAM usage and may slow down training)"
     ),
 )
+@click.option("--max-seq-len", type=int)
+@click.option("--max-batch-len", type=int)
 @click.option(
-    "--model-name",
-    default="instructlab/merlinite-7b-lab",
-    show_default=True,
-    help="model name to use in training",
+    "--effective-batch-size", type=int, help="total batch size across all GPUs"
+)
+@click.option(
+    "--save-samples",
+    type=int,
+    help="how many amples to process before saving a checkpoint",
+)
+@click.option("--learning-rate", type=float, help="learning rate for training")
+@click.option("--warmup-steps", type=int, help="warmup steps for training")
+@click.option(
+    "--deepspeed-config",
+    type=click.Path(),
+    help="configuration to use for deepspeed training",
+    hidden=True,
+)
+@click.option(
+    "--cpu-offload-optim", type=bool, help="if true enables optimizer offload"
+)
+@click.option(
+    "--cpu-offload-params", type=bool, help="if true, enables parameter offload"
+)
+@click.option("--ds-quantize-dtype", type=click.Choice(["nf4", "fp8"]), default=None)
+# below flags are invalid if lora == false
+@click.option("--lora-rank", type=int, help="rank of update matricies")
+@click.option(
+    "--lora-alpha", type=float, help="how influential/strong lors tune will be"
+)
+@click.option("--lora-dropout", type=float, help="dropout for LoRA layers")
+@click.option(
+    "--target-modules",
+    type=str,
```

Comment:
maybe this should be hidden since it's pretty implementation-specific?

Reply:
I think a first pass where all flags are visible is best, then we can widdle away